diff -aur sources/xorg/git/xorg/proto/x11proto/Xosdefs.h.001 sources/xorg/git/xorg/proto/x11proto/Xosdefs.h
--- sources/xorg/git/xorg/proto/x11proto/Xosdefs.h.001	2013-02-10 19:39:41.332374071 -0500
+++ sources/xorg/git/xorg/proto/x11proto/Xosdefs.h	2013-02-10 19:57:02.292380459 -0500
@@ -70,6 +70,9 @@
 #  endif
 # endif
 
+#ifdef __ANDROID__
+typedef unsigned long fd_mask;
+#endif
 
 # ifdef __APPLE__
 #  define NULL_NOT_ZERO
diff -aur sources/xorg/git/xorg/proto/x11proto/Xos_r.h.001 sources/xorg/git/xorg/proto/x11proto/Xos_r.h
--- sources/xorg/git/xorg/proto/x11proto/Xos_r.h.001	2013-02-10 19:39:41.332374071 -0500
+++ sources/xorg/git/xorg/proto/x11proto/Xos_r.h	2013-02-10 19:57:02.292380459 -0500
@@ -323,17 +323,18 @@
 typedef struct {
   struct passwd pws;
   char pwbuf[X_LINE_MAX];
+  struct passwd* pwp;
 } _Xgetpwparams;
 # if defined(_POSIX_REENTRANT_FUNCTIONS) || !defined(SVR4)
 #   define _XGetpwuid(u,p) \
-((getpwuid_r((u),&(p).pws,(p).pwbuf,sizeof((p).pwbuf)) == -1) ? NULL : &(p).pws)
+((getpwuid_r((u),&(p).pws,(p).pwbuf,sizeof((p).pwbuf),&(p).pwp) == -1) ? NULL : &(p).pws)
 #   define _XGetpwnam(u,p) \
-((getpwnam_r((u),&(p).pws,(p).pwbuf,sizeof((p).pwbuf)) == -1) ? NULL : &(p).pws)
+((getpwnam_r((u),&(p).pws,(p).pwbuf,sizeof((p).pwbuf),&(p).pwp) == -1) ? NULL : &(p).pws)
 # else /* SVR4 */
 #  define _XGetpwuid(u,p) \
-((getpwuid_r((u),&(p).pws,(p).pwbuf,sizeof((p).pwbuf)) == NULL) ? NULL : &(p).pws)
+((getpwuid_r((u),&(p).pws,(p).pwbuf,sizeof((p).pwbuf),&(p).pwp) == NULL) ? NULL : &(p).pws)
 #  define _XGetpwnam(u,p) \
-((getpwnam_r((u),&(p).pws,(p).pwbuf,sizeof((p).pwbuf)) == NULL) ? NULL : &(p).pws)
+((getpwnam_r((u),&(p).pws,(p).pwbuf,sizeof((p).pwbuf),&(p).pwp) == NULL) ? NULL : &(p).pws)
 # endif /* SVR4 */
 
 #else /* _POSIX_THREAD_SAFE_FUNCTIONS */
diff -aur sources/xorg/git/xorg/lib/libXfont/src/util/fontxlfd.c.001 sources/xorg/git/xorg/lib/libXfont/src/util/fontxlfd.c
--- sources/xorg/git/xorg/lib/libXfont/src/util/fontxlfd.c.001	2013-02-10 19:58:16.792354452 -0500
+++ sources/xorg/git/xorg/lib/libXfont/src/util/fontxlfd.c	2013-02-10 19:59:33.792373698 -0500
@@ -40,7 +40,7 @@
 #include	<X11/Xos.h>
 #include	<math.h>
 #include	<stdlib.h>
-#if defined(sony) && !defined(SYSTYPE_SYSV) && !defined(_SYSTYPE_SYSV)
+#if (defined(sony) && !defined(SYSTYPE_SYSV) && !defined(_SYSTYPE_SYSV)) || defined(__ANDROID__)
 #define NO_LOCALE
 #endif
 #ifndef NO_LOCALE
diff -aur sources/xorg/git/xorg/xserver/hw/xfree86/Makefile.am.orig sources/xorg/git/xorg/xserver/hw/xfree86/Makefile.am
--- sources/xorg/git/xorg/xserver/hw/xfree86/Makefile.am.orig	2013-03-09 19:24:19.000000000 -0500
+++ sources/xorg/git/xorg/xserver/hw/xfree86/Makefile.am	2013-03-09 19:26:11.502374978 -0500
@@ -28,9 +28,9 @@
 SUBDIRS = common ddc x86emu $(INT10_SUBDIR) os-support parser \
 	  ramdac $(VGAHW_SUBDIR) loader modes $(DRI_SUBDIR) \
 	  $(DRI2_SUBDIR) . $(VBE_SUBDIR) i2c dixmods \
-	  fbdevhw shadowfb exa $(XF86UTILS_SUBDIR) doc man
+	  android shadowfb exa $(XF86UTILS_SUBDIR) doc man
 
-DIST_SUBDIRS = common ddc i2c x86emu int10 fbdevhw os-support \
+DIST_SUBDIRS = common ddc i2c x86emu int10 android os-support \
                parser ramdac shadowfb vbe vgahw \
                loader dixmods dri dri2 exa modes \
 	       utils doc man
diff -aur sources/xorg/git/xorg/xserver/configure.ac.001 sources/xorg/git/xorg/xserver/configure.ac
--- sources/xorg/git/xorg/xserver/configure.ac.001	2013-02-10 20:04:08.782334611 -0500
+++ sources/xorg/git/xorg/xserver/configure.ac	2013-02-10 20:08:38.272377974 -0500
@@ -408,7 +408,7 @@
 	AC_DEFINE(PCVT_SUPPORT, 1, [System has PC console])
 	AC_DEFINE(WSCONS_SUPPORT, 1, [System has wscons console])
 	;;
-  *linux*)
+  *jjlinux*)
 	DRI=yes
 	;;
   *solaris*)
@@ -1694,7 +1694,7 @@
 	dnl ===================================================================
 
 	case $host_os in
-	  linux*)
+	  jjlinux*)
 		if test "x$LNXAPM" = xyes; then
 			XORG_CFLAGS="$XORG_CFLAGS -DXF86PM"
 		fi
@@ -2257,6 +2257,7 @@
 hw/xfree86/exa/man/Makefile
 hw/xfree86/fbdevhw/Makefile
 hw/xfree86/fbdevhw/man/Makefile
+hw/xfree86/android/Makefile
 hw/xfree86/i2c/Makefile
 hw/xfree86/int10/Makefile
 hw/xfree86/loader/Makefile
